src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/service/impl/CaseFormatterServiceImpl.java

```
@@ -50,7 +58,7 @@ public DivorceSession transformToDivorceSession(CoreCaseData coreCaseData) {
     }
 
     @Override
-    public CoreCaseData addDocuments(CoreCaseData coreCaseData, List<GeneratedDocumentInfo> generatedDocumentInfos) {
+    public Map<String, Object> addDocuments(Map<String, Object> coreCaseData, List<GeneratedDocumentInfo> generatedDocumentInfos) {
 
         if (coreCaseData != null && CollectionUtils.isNotEmpty(generatedDocumentInfos)) {
```

can coreCaseData be `null`?

If not, why don't we add:

```
if (coreCaseData == null) {
    exception - invalid data
}
```

?

---

CoreCaseData should never be null. Wondering if I should throw a custom exception or let the code throw an NPE when it tries to use coreCaseData.

---

Up to you. I have already approved the code, but I think we shouldn't be so defensive when we know that null must never happen here. This can actually hide some bugs.